### DISCUSSION TITLE
Shield Design V2

### DISCUSSION THREAD

**Discussion by DeeEmm** [DIY-Flow-Bench/DIY-Flow-Bench#129] [Shield Design]
<img width="2032" alt="Screenshot 2023-02-22 at 1 37 35 pm" src="https://user-images.githubusercontent.com/3038710/221056511-c2811c96-1b22-46d7-acb5-e2327becc4a0.png">

Shield Version 2.0 is now being manufactured. This incorporates updates to address all of the issues found in the original shield.

I have also made the Shield project public should anyone be interested. This can be accessed via the project link above or directly [HERE](https://github.com/users/DeeEmm/projects/3/views/1)

I have already discovered some things that need to be addressed as in my haste I forgot to add footprints for the XGZP6899A style of pressure sensors, which I only received the other day. These are a cheaper alternative to the MPXV7007DP but unfortunately do not share the same pin outs. I've already updated the PCB design to include the new footprint but will hold off on getting another batch of boards made until I can test the first batch.

There are five V2.0 PCBs in this first batch. Assuming that my initial testing goes well, I'll make the remaining four bare PCBs available to those who previously expressed an interest in beta testing.

I'll update my progress with the V2.0 boards here rather than in the main monthly update thread, to seperate code and hardware discussions and avoid thread pollution.

## Known Issues

Just going to keep track of issues found here for future reference. These only apply to this specific PCB version, which is identifiable from the DIYFB V2.0 screen printed onto the board.

### Non-breaking issues

- Pinouts for Sparkfun PCA9306 I2C level converter on PCB does not match the Sparkfun BOB that I have. Having now checked the Footprint that I used, I can see that I did use what is supposed to be the Sparkfun footprint, however it is different from the physical board I have. Not sure if this is a version thing or perhaps an error from Sparkfun. The issue is easily overcome by using PCB Jumper Cables. Whilst not ideal, at least it works. I will review and update for V2.1 as the ideal scenario is that all of the BOBs are plug and play
- Pinouts for BME280 are also different from the PiicoDev board that I am using. However the original footprint I chose was for the generic BME boards that you find all over the internet. I did not choose to use the PiiconDev layout as it is not so widely available. The caveat here is that most of those generic boards appear to be BMP280s and not BME280s. Again I just used jumper cables to solve my issue. I need to review this and determine what specific pin layout I should use. I did originally intend to add several pin layouts but this does not play nice with the auto-router. Again the ideal solution is that all BOBs are plug and play.
- Reviewing the schematic I note that for some reason I've given the BME280 a 5v supply. Looking through the tech specs it looks like most BME280 BOBs are 5v tolerant with 3.3v shifted outputs but I will correct this for the final version.

### Breaking issues

- None so far :)

**Comment by DeeEmm** [comment]
I'm sharing the schematic for the shield design here. 

[Schematic_DIY-FLOW-BENCH-V2_2023-02-22.pdf](https://github.com/DeeEmm/DIY-Flow-Bench/files/10819441/Schematic_DIY-FLOW-BENCH-V2_2023-02-22.pdf)

### Changes from V1.0 PCB

I've changed Molex style connectors to IDC style connectors. These are easier to source and also allow the use of pre-made IDC cables (either ribbon or round) which should simplify wiring.

I2C logic level shifter is installed to allow A to D converter to run at 5 volts. This is required for MAF and Pressure sensors which are all 5v devices

I doubled up on the Darlington array channels to increase output current. This seemed like a no brainer as the spare channels were redundant. The Darlington array provides 12V control for the Vac motor switching and is intended to be used with 12v relays. These can then be used to switch a 240v load. There are three vac channels.

I added analog outputs for VFD and Bypass valve control. This is directly from the ESP32's DAC channels which can provide 0-3v at 8 bit resolution. Most programmable VFDs have the option for input scaling and should be able to work with a 0-3v signal. However if your VFD requires some other range (e.g. 0-10v) then external signal conditioning may be required to convert the 0-3v signal to whatever signal range is required.

At this stage I have elected not to fit a hardware based frequency to voltage converter as my understanding of the ESP32 / ADS 1115 input specification is that it should be able to handle frequency style MAF sensors directly. I need to validate this but currently do not have a frequency based MAF to test with. Frequency based MAF support has been pushed out until V3 code but is still a consideration when designing the PCB. On paper it should work, but real life testing is required.

I have added additional footprints for the XGZP6899A pressure sensors. These are omitted from the V2.0 PCB but will be present in subsequent versions.

An (optional) SD card has now been added in readiness of V3 code development. The intent for this is storage of flow test data. The intent is to create a JSON based file for each test object, which can have a maximum of 12 data points. This equates to 50 thou increments with a max lift of 600 thou, which should be ample for all but the most extreme projects. Valve lift increments can also be manually changed so you can use thou or millimeters with whatever period you require. A fallback function is also being added to allow data to be downloaded directly from the browser.

**Reply by black-top** [reply]
Just a few thoughts to ponder. 

I think it would be a good idea to provide components on the board for signal conditioning in the future. For example, a transistor with an op-amp , configured with several different resistor dividers for different output voltages 0-5/0-10V, which could be choose with a simple jumper on the pcb.

Dc dc buck converter output relatively noisy it may need additional filtering for analog sensors with high resolution ADC.  Maybe in this case, a simple analog regulator like lm7805 is better suited, for which the required voltage can be set with a simple divider. For very stable voltage requires more complex schemes, for example with an op-amp but perhaps it is not necessary.  Additional capacitors close to each component might be useful too. Another source of noise can be Vac motors, they cause a lot of electrical noise, especially when using triac regulators without zero-crossing.

Another additional digital input for the encoder can be useful for connecting a swirl meter.

**Comment by DeeEmm** [comment]
I hadn't thought of the swirl meter, a good idea, we should be able to incorporate this on the existing ADS1115 ADC instead of either the MAF or differential pressure sensors as only one of these is used at a time. I've added it to the road map. No idea at this stage how that might look but lets capture it for later analysis.

Adding additional signal conditioning could be a good idea. It's hard to know how far to go. At the moment I'm trying to take the most simplistic approach, which is good if it proves to be repeatable and reliable. My philosophy is to determine each functional aspect and then cater for that in a modular and easy to build manner so that it's easy for the end users to build. Adding transistors and op amps that require some degree of setting up or validating is going to be way outside of the skill sets of most users and is something that is more suitable to a pre-built solution or perhaps a later added-on or hack for those who like to tinker. A small 'prototyping' area is something that I have seen on other boards, I think I've got a small bit of real estate left, I'll look to add one for this purpose.

To a degree, the nature of the official shield is to be simple and easy to assemble whilst still also providing _good_ results. Note that I say good not perfect. Our goal is only to _product results comparable with a professional bench_. There may be opportunity to improve the design and create much better shields down the track, and this is actually something that I hope that the community does, but it's hard to determine where that 'line in the sand' is for this initial basic version. If I can get this working with a minimal component count and produce reliable and accurate results, then I think that is the goal achieved. 

Just on this, the key thing to ponder here is what do we mean when we say 'accuracy'. Off the top of my head and thinking aloud, I think that results within 1 cfm are likely good enough for most users, especially when you consider that most traditional benches are using a printed lookup tables to 'calculate' results from human read manometers (which includes 'human error' and does not compensate for environmental conditions). We will easily be inside of the tolerance of a traditional bench, which is the basic project goal achieved. You will note that in comparison our code is working down to two decimal places, so I think that we will easily exceed this without even trying.

How much of an impact will some ripple or spikes have on this end result is hard to quantify, but I expect that as the output is many orders of magnitude larger than these inputs, they will essentially have minimal impact (I suspect much less than 1 cfm). I do not recall seeing any deviation larger than this in testing so far. Of course without specifically measuring this, it is all just conjecture and is nothing without further testing, which is what we need to do next. I've added a note to the PSU testing to look at this.

Regarding filtering, there is already some basic filtering on each of the pressure sensors, this is the same as other automotive projects that I have designed / used. In those cases it works well and those are pretty noise environments. Similarly there is also filtering on the MAF and high speed inputs, but again this is also only basic R/C filtering and component values may need to be tuned. The breakout boards also contain basic signal conditioning on board so there is no need to include this on the main PCB as it is specifically designed to be used with the BOBs.

I believe that the buck style converters should give a supply that is within spec, (see my [other comment](https://github.com/DeeEmm/DIY-Flow-Bench/discussions/122#discussioncomment-5111411)) but it is also possible to add a discrete voltage regulator and associated componentry within the same footprint so that either one can be utilised. This would satisfy both approaches and is something that needs to be determined by testing. Once I get the new board assembled and running I'll do some probing with the scope and see what things look like. If the buck converters are not suitable then I'll look to swap them out, but so far the results I have seen are okay (acceptable?). I'm definitely open to incorporating something like the lm7805 if required. I might try to breadboard a comparison test if the LM2596's are not quite suitable.

Vac motor control is all taken care of externally. There is no intention of bringing noisy supply lines anywhere near to the PCB. I did have thoughts on adding some opto-isolators to the board but these can also be added externally, and if using the off-the-shelf Arduino compatible relay boards, are already present. Of course relying on the end user not to just hook up any old relay should really push me towards including them on the PCB. I should probably take a look at this and have added it to the roadmap. I think existing filtering should catch anything else, but lets take a further look and see. We do of course need to quantify and validate everything with actual testing.

All componentry is of course to be finalised, as-is the final shield design and I suspect that there will still be one or two more iterations in the shield design before it is finalised. Reality is that we still do not have a shield that is 100% functional. Hopefully when the new PCBs arrive this goal will be one step closer.

/DM

**Reply by black-top** [reply]
I have come across swirl meters made from a rotary encoder with a propeller similar to what rc models use. A similar principle works in a flow cap on a dyno. I think a more suitable option would be digital input to calculate encoder pulses rather than adc. 

1Cfm resolution would be great. Here are just a few thoughts on what can cause unstable measurement values. Measurements will be influenced by many factors at the same time, so it may be difficult to identify them. Electrical noise is one of the options that can cause it. If the device is not powered by batteries, noise from the mains may be transmitted. It will be seen when we start real testing, maybe there will be no problem with that.

When the final version of this code is available, I think I will design my shield the most suitable for my needs, because as you mentioned, a lot of things can be loaded on the current version, although these may be unnecessary functions for most.

Another observation relates to pressure regulation. David V izard in his books advocates for floating pressure flow bench as a more accurate method because it simulates a more real world situation. It will be interesting to experiment and compare the results without adjusting the pressure, when the program automatically calculates the data to the standard 28'' h2o pressure in real time. Perhaps the pressure regulation will not even be necessary.

**Comment by DeeEmm** [comment]
Ahh yes of course, the swirl would require a rotary encoder. This would require two digital inputs. I'll take a look at the I/O Schedule and see where it can be fitted in. As you say there's lots of features and it is unlikely that all will be used at once, this allows some I/O to be reused. If we can cater for all aspects in code then the end user can pick whichever suits their needs.

Regarding mains noise, the PSU should filter the mains noise out. This will rely on using a good quality 240v to 12vDC PSU. Of course it is also be possible to run the unit from a 12vDC car battery as well. 

Pressure regulation is an interesting topic. I too read David Vizards book and agree with his methodology. This is why the basic hardware design in the WIKI outlines a simple shop vac setup. The main constraint is that there is a ceiling to what the vac source can supply. At some point the test piece will outflow the vac source and measurements will plateau. For this reason there is a simple validation check so that when the flow values approaches 90% of the unrestricted flow of the vac source a warning will be displayed. This warning becomes more appropriate when using the bench in a floating reference pressure mode. Normally the user will notice that the limit has been reached as they will not be able to achieve the reference depression at the given lift, but when you are not setting the depression you will miss this.

Converting the reference depression is of course a sticking point for many.  Even with real-time conversion many will argue that the test is not valid or results are not accurate because a different reference depression is used. So it will definitely be interesting to be able to run comparison testing. 

I think for many however the believability of results from the DIYFB project will hang on convincing people, using results made with the constraints of a regulated reference depression setup, and comparing them head to head with a professional bench. Apples to apples, so to speak. But even then there are many that will not accept that a $300 bench can be compared to a $12000 professional bench, because how can it? For a genuine positive results based outcome it will take a lot of testing, a lot of data and a lot of convincing.

**Comment by DeeEmm** [comment]
Ok, so the speed sensor pin + spare pin are most logical for use with external encoder. The speed sensor is specifically for things like turbine shaft speed which was a request from an interested party. Also the BCD orifice coding pins are free on MAF benches, or the Serial 2 pins if an external gauge is not being used. Essentially any of those inputs can be easily reconfigured by remapping the definitions in the pins.h file. 

Unfortunately the ESP32 is very I/O limited and does not allow for all permutations to be hard wired so there is going to be some crossover between features / GPIO assignments.

I guess this also shows that there is a benefit to limiting the signal conditioning on the generic / official shield, as it does not tie the inputs to a specific usage and allows them to be easily reassigned. It is of course possible to add additional signal conditioning should it be needed. I've already laid out a pretty large prototyping area for this purpose.

I also had some further thoughts on the DC-DC converters / Buck PSUs. My reason for using two seperate bucks and not simply relying on one with a voltage divider was to seperate the onboard bus voltage (3.3v) from the field bus voltage (5v). Each buck has some level built in filtering / smoothing and so the separation provides segregation between the different bus potentials which somewhat addresses the issue of noise transfer.

I also think that being able to individually dial in each bus voltage is important from an ease of use perspective. Using voltage dividers will not result in an accurate voltage level unless high precision matched resistor pairs are used. This just adds complexity for both component sourcing and commissioning and puts it outside of the skill set of most end users. Using a buck with a multi turn pot is pretty straightforwards and easily achievable by most.

Hopefully I'm achieving the right balance. It's kinda hard to know for sure, but my experience with the DIY / maker community is that there are a lot of people who can solder a project like this, but not so many who can fault find / diagnose issues. So it really needs to be as fool-proof as possible, otherwise I will end up spending my days running tech support lol.

**Comment by DeeEmm** [comment]
Here's the shield with the added prototyping area. It's just a grid of Vias that can be used to solder components to. Basically like a breadboard. 

<img width="1871" alt="Screenshot 2023-02-27 at 6 59 54 pm" src="https://user-images.githubusercontent.com/3038710/221514379-48f6709d-5923-42cb-813c-b2e9a40ef902.png">

I've brought the inputs mentioned previously to the Vias at the left side of the grid along with ADC3 which is the Pitot sensor. These are basically all of the inputs that are non-essential to the core operation. I've also added a rail of 3.3v and 0V Vias too. This should pretty much accommodate any hacks or mods that people want to do whilst still retaining basic stock functionality of the shield for those who just want to use it as-is.

<img width="1347" alt="Screenshot 2023-02-27 at 7 08 05 pm" src="https://user-images.githubusercontent.com/3038710/221514408-3e386a52-8e0c-46cf-a446-47cec16d3922.png">

You can also see that I've added the additional pads for the XGZP pressure sensors to the underside of the board. As a result of this I had to add an additional alternate mounting location for the SD card. This way if the pressure sensor locations on the bottom are populated then the SD card can be mounted to the top and vice-versa.

Pondering further on the PSU discussion, I've decided that at present I'm not going to do anything further with regards to incorporating something like an lm7805. So instead of adding additional circuitry to the board to support an alternate PSU setup, this can easily be added as a substitute daughterboard in place of the buck converters. This more closely follows the general philosophy of the board and means that if you want to build a shield with better power management you should still be able to do this using the basic shield.

Hopefully what we will then end up with is the existing 'easy-to-build' setup using off the shelf buck converters and those who want to run a different (better?) solution can simply substitute the buck converters for a custom daughterboard. The end result might be a difference in system accuracy / quality but we can quantify what that might look like with further testing.  This also means that I don't  get too far off track with what I'm doing otherwise this may never get finished lol.

**Comment by DeeEmm** [comment]
As posted elsewhere I've now assembled the shield PCB and am working through testing. So far the shield is working as expected. The pin layout for the PCA9306 level shifter differs from the Sparkfun BOB I purchased but that was easily addressed with jumper leads. 

For far the PSU, BME, ADC and PCA circuits are tested and working.  I still need to perform tests for pressure sensors, steppers and darlington array.

![PXL_20230413_025536457 MP](https://user-images.githubusercontent.com/3038710/231636672-d0298b9f-226d-4a58-b848-9705e597321d.jpg)

I have some spare boards that I will send out to the early adopters / contributors once I get the necessary documentation finished.

So far, So good !!

**Comment by black-top** [comment]
I'm ordering components suitable for this pcb and I can't find PCA9306 V2, as far as I can see, the v1 version do not have an enable pin, apart from the footprint and pinout looks same, but enable is not needed according to the schematic, is there anything else specific that would make this version better in this case? I use transistor-based lever shifters in my setup, everything seems to work fine. 
What recomended sd card module or just any 3.3V version?

**Reply by DeeEmm** [reply]
The enable for the PCA9306 V2 is tied to the 5V pin. It performs no functional purpose for this project

The only thing I can think to check is the change log / release notes to find out what is different between V1 and V2.

Assuming they are functionally identical there should be no reason that the earlier version will not work. Unless of of course there is some issue with it which is what spurned the revised design. This is not uncommon with these small run BOBs.

Not sure if you noted but the device needs to be connected by jumpers as the pin outs are not the same, so you could actually use ANY PCA9306 BOB

**Reply by black-top** [reply]
If something doesn't work, the only problem is the long delivery time, but I think everything should work well with a transistor-based level shifter, as in my current setup, just have to be connect with wires, because the pinout is different. PCA9306 probably doesn't have a serious advantage.

**Reply by DeeEmm** [reply]
Indeed. It should theoretically work.

I guess you can take a look at the I2C standard and check what the logic levels are and then make sure they fall in the same range. IIRC the lines are either high or low (so Vcc = 3.3v OR 5v). The pull-up resistors are already in place on the shield for both sides of the I2C bus, you can always tweak the values if necessary.

**Comment by DeeEmm** [comment]
Reviewing the schematic I note that for some reason I've given the BME280 a 5v supply. Looking through the tech specs it looks like these BOBs are 5v tolerant with 3.3v shifted outputs. The critical aspect here is what voltage the ESP is seeing as they are not 5v tolerant. 

The BME is working fine on my system with the 5v supply, but I will update this on the final version and change it to 3.3v

if you are testing with one of these boards and are concerned then change the 5v pin to 3.3v. In most cases the BME280 will need to be jumpered at any rate.

**Comment by danielerpl** [comment]
Hi, there a possibility to buy the PCB ? without components

**Reply by DeeEmm** [reply]
Hi @onzio88r 

I have one left from the initial prototype run. The caveat with this is that it is only available if you can get get involved in the testing.

At this stage of the project I need to get data from several different builds so that I can validate the design and code. Without this feedback there is no way to measure the accuracy of the design or address any shortcomings, so it is a critical part of the development process.

At the moment the rest of the boards have been sent out and I am waiting for the recipients to integrate the controllers into their benches and start to run some tests and provide some data.

If you can commit to doing the same I'm happy to send you out a board.

/DM

**Reply by danielerpl** [reply]
Absolutely i can commit everything to you, obviously after tour check!! 
because i have also in some features to integrate in your main project.

i am making my own bench and i need some information and functionalities, if you are happy to collaborate it is very appreciated!! At the moment as you know, i am still configuring the project to be up and running but without the board it’s a bit hard! 
so, my bench it’s ready, i would like to connect the board now to start making some test and of course give to you feedback and results

let me know what you need by my side to start the process to have the board 

Many thanks, very appreciated 
Daniele

**Reply by DeeEmm** [reply]
Always happy for collaboration / contributions, development of a project like this is much easier when there's more than just one contributor :)

Flick me an email with your address to deeemm@deeemm.com and I'll post the board to you.

What type of functionality do you need for your bench? I have tried to include most possibilities for bench control in the design, but some of those features are only on the roadmap at this stage.

The V3 roadmap will give a good general overview of the current direction and the additional features currently considered.

https://github.com/users/DeeEmm/projects/4

Once the current V2 version has been validated then I will start to take a look at the V3 features. If there are specific features that you need that are not already there we can also add those to the roadmap.

**Comment by danielerpl** [comment]
[](url)
![IMG_4557](https://github.com/DeeEmm/DIY-Flow-Bench/assets/14088795/48f2b0c4-49ad-4248-bf36-174048aaa4eb)

Some parts arrived and soldered!! 
@DeeEmm  Which type of power supply do you suggest ?

**Reply by Ronnie180** [reply]
Wall wart or the black square ac/dc power supplies is what I suggest if I'm reading that right..

**Reply by DeeEmm** [reply]
Good stuff. :) 

Yes a 12V DC wall wart or switched mode PSU will work. 

If you plan to integrate the steppers you will need to make sure that the amperage is high enough. The 8825 stepper drivers will generally supply a max of 2.5 amps. So if you want to run both the flow control and auto valve adjuster you will need at least an 8 amp PSU (2x2.5A + 3A for the ECU) which generally means you are looking at a standalone switched mode PSU rather than a wall-wart. 

If you are not using the steppers then a 12V 3A PSU should work. 

One consideration with wall-warts is that the are often quite crude and can be very (electrically) noisy. This can lead to issues with sensor signals. A higher quality PSU with some form of output filtering is the best choice if available.

**Reply by Ronnie180** [reply]
I had wondered about the wall warts since they are kinda cheaply made & throwed together plus thousands of them out there to, so it's good to confirm that.  Is there a particular switched mode psu you recommend since its lots of them out there to & Since I have 15 or 20 steppers different sizes on the shelf, might as well put 2 of them to use & go full automation, so more than likely I'll use 2 but I can see some just using one or the other as well to.

**Reply by DeeEmm** [reply]
I typically use the cheap units available online. Something like this...

https://www.digikey.com.au/en/products/detail/mean-well-usa-inc./LRS-100-12/7705004

You can find them cheaper on ebay and elsewhere. It has been mentioned before (by others) that they are still not the best from a noise perspective, but I've used them a lot on Arduino / ESP projects with no real issues. Definitely a lot better than the wall warts and not bad from a $ / amp perspective. The only consideration is that as they have exposed terminals, they ideally need to be mounted in some kind of enclosure to prevent straying fingers from touching the terminals. but packing up the controls into some kind of box or enclosure is recommended anyhows.

**Reply by Ronnie180** [reply]
Okay. Thanks.

**Comment by Ronnie180** [comment]
Got my vacuum motors today. Ametek 119996. Flow 82 cfm, 5.3 amps at 44" vacuum.

**Comment by Ronnie180** [comment]
Just in case anybody is looking for the vacuum motors I mentioned which flow around 15cfm/amp & are 220-240 volt motors, single stage flow through & measure 6 inches wide by 5-1/2 inches long & 1-1/2 inch inlet hole. 6 Pack shipped to me was $182.00 which is hard to beat from what all I've looked around for. They are 2 yr old stock but brand new still in the original boxes & sealed.

Here the link:   https://www.ebay.com/itm/374208029133?hash=item57208961cd:g:Z6gAAOSwTDdi8sJE

**Reply by DeeEmm** [reply]
Looks like a reasonable deal but shipping kills it for me here in Aus.

**Reply by Ronnie180** [reply]
Yeah I looked to see what shipping would be & It's the same as the cost of motors, little to much for that

**Comment by danielerpl** [comment]
Nice!! I will get one 12v online, at the moment I think I don't need the stepper motor but I will consider to use it later 
Thanks Guys

**Reply by Ronnie180** [reply]
You welcome, Mick is the go to man.

**Comment by danielerpl** [comment]
Guys, I need help! I have some problem to find the component SPARKFUN PCA9306 V2
please can you share a link from a website in Europe. I found one but the pinout it's differente 
This is what I found 
https://www.robot-italy.com/it/sparkfun-level-translator-breakout-pca9306.html

**Comment by danielerpl** [comment]
I have also another question @DeeEmm for the lm2596s which is the output to set ? 
I need info for the U8 and U9 component
It is explained in a document here ?

**Comment by DeeEmm** [comment]
The LM2596 are common power bucks and may be substituted with other comparable power bucks. They are easily found on ebay. Here is an example - https://www.ebay.it/itm/165082764044 I recommend to use an adjustable version so that you can dial in the voltage accurately.

To set the voltages of the the LM2596 power bucks I have been measuring at the output of the device between the Out- and Out+ pins. The bottom buck is 3.3v and the top buck is 5v. This is labelled on the board.

The PCA9306 that you link to is correct. On this version of the board the pin locations are incorrect. You will need to attach the  PCA9306 BOB via jumper leads. This will be fixed in the final version of the PCB.

**Reply by danielerpl** [reply]
Nice, about the LM2596 I read the value on the board, and for double check I preferred your answer to be sure! 
about the PCA9306 ok, I will use some jumper leads to connect it and respect each name from the PCA9306 board to your PCB board 
Many thanks

**Reply by danielerpl** [reply]
I think I found the correct one. This one has the pinout like your PCB board 
@DeeEmm  can you confirm ? 


https://www.amazon.it/Hailege-PCA9306-Translator-Breakout-Bidirectional/dp/B0B778Z6RC/ref=sr_1_1?__mk_it_IT=ÅMÅŽÕÑ&crid=3MJMBIJJYDKOZ&keywords=SparkFun+Level+Translator+Breakout+-+PCA9306&qid=1697186304&sprefix=sparkfun+level+translator+breakout+-+pca9306%2Caps%2C118&sr=8-1

**Reply by DeeEmm** [reply]
I've not used this part but it does look to be the same pin-outs.

**Reply by danielerpl** [reply]
I will try this one, I prefer not use the jumpers on the board. This is a very basically component, should be not a problem

**Comment by danielerpl** [comment]
@DeeEmm 
Another little question, I have some problem to fin on the market the capacitor 10nF for the designator C2.C5.C19 
Can I replace it with one from this box ? 
Please can you share a link where you buy capacitor? It will be very appreciated 

![IMG_4703](https://github.com/DeeEmm/DIY-Flow-Bench/assets/14088795/5aafd9ae-f47b-464e-8aed-7730dc0a49e5)

**Reply by DeeEmm** [reply]
10nf is the same as 0.01uf try to search for that instead. Some vendors list nf and some uf.

If you cannot find one then you might get away with the 0.1uf. It is just a filter on the vcc supply to the sensor. It should not affect the reading.

**Reply by danielerpl** [reply]
I am asking you because it's very hard to fin a 10nf or 0.01uf 
This is why I sent you what I have 
I will try to fin the perfect one, otherwise.....I have to use the 0.1uf

**Comment by danielerpl** [comment]
@DeeEmm just a question about the resistor. The number R8, is this the same of R1,R2....... 
because the value in name is 0 k ohm but in the manufacturer part is the same one 
I am looking in this document 

BOM_DIY-FLOW-BENCH-V2.pdf

**Reply by DeeEmm** [reply]
Yes this is actually a wire link. I just borrowed the resistor symbol to be able to insert the pads in the design. Just use a wire link instead of a resistor. The schematic should also be labeled as "link"

**Reply by DeeEmm** [reply]
(I think the reason I did this was to be able to break out the MAF signal in case people wanted to bypass the ADC and connect an external circuit instead)

**Reply by danielerpl** [reply]
Perfect, I will put there a wire link 👍🏼

**Comment by danielerpl** [comment]
About the WeMoss D1, I bought the board and I need a suggestion to put it on the board. Have you soldered the board on the pub ? Because I was thinking to use the wire to connect it to the board 
I think will be much better with wires. Just in case I have any problem with the WeMoss board, I can replace it quickly

**Reply by DeeEmm** [reply]
Sorry, I missed this post.

I use pin headers on the board and pin sockets on the PCB, This allows the ESP32 to be unplugged and easily swapped out if there is an issue. You will also note that there are some components mounted to the PCB underneath the ESP32, so to be able to service these components it is better if the board can be unplugged.

![PXL_20231130_025750432](https://github.com/DeeEmm/DIY-Flow-Bench/assets/3038710/e7793124-801a-4d35-a19a-547e36308536)

![PXL_20231130_025807300](https://github.com/DeeEmm/DIY-Flow-Bench/assets/3038710/168a3ca7-5332-493d-b77e-d21d4e6b6c11)

**Comment by danielerpl** [comment]
Don’t worry is not a problem.
Good idea to have the board below the PCB! Nice!
I this i will use external wires to connect the esp32 to the pcb.
Just another question about power, are you powering both? Pcb and esp32
with the dc connector? I bought your same esp32 board

Thanks a lot

**Comment by danielerpl** [comment]
@DeeEmm board completed, up and running....there is no fire and I thinks it's all good :D 
So, I am trying to upload the last released code on the WeMod D1 board but I have a problem. 

This is the error : 

A fatal error occurred: This chip is ESP8266 not ESP32. Wrong --chip argument?

I am using this configuration in the platformio.ini 

[env:release]
build_type = release
; board = esp-wrover-kit
;board = esp32dev
board = wemos_d1_uno32
; board = ArduCAM_ESP32S_UNO
upload_protocol = esptool
; upload_speed = 921600
upload_speed = 460800
lib_ldf_mode = chain
lib_deps = 
	bblanchon/ArduinoJson@^6.19.4
	esphome/AsyncTCP-esphome
    esphome/ESPAsyncWebServer-esphome
	https://github.com/terryjmyers/ADS1115-Lite.git
	; https://github.com/UncleRus/esp-idf-lib
	https://github.com/fabyte/Tiny_BME280_Arduino_Library.git
	majicdesigns/MD_REncoder@^1.0.1
lib_ignore = 
	; SPI

Do you have any suggestion where I have to make changes ? 
Many thanks 

![tempImagewysGMl](https://github.com/DeeEmm/DIY-Flow-Bench/assets/14088795/7c6103fa-ad22-4f97-a1eb-edb2dd0be1dc)

**Reply by DeeEmm** [reply]
It does look like your D1 has an 8266 processor - look at the markings on the processor shield - ESP8266MOD

**Reply by danielerpl** [reply]
@DeeEmm  exactly, I tried to change some config but I have the same error. Do you have any suggestion where to put the correct processor config ? 
Many thanks for the support

**Reply by danielerpl** [reply]
@DeeEmm  I solved the problem about board config, I used what explained in the official guide 
https://docs.platformio.org/en/latest/boards/espressif8266/d1.html?utm_source=platformio&utm_medium=piohome

But I am trying to solve now some dependencies about the libraries 
Can you try to build the project with this platformio config ? 
Just to help me on soling quickly the issues. 
I will share you what I did when I completed all errors
